app/models/link_checker_api_report/link.rb

```diff
@@ -0,0 +1,16 @@
+class LinkCheckerApiReport::Link < ActiveRecord::Base
+  serialize :check_errors, Hash
+  serialize :check_warnings, Hash
+
+  belongs_to :report, class_name: LinkCheckerApiReport
+
+  def self.attributes_from_link_report(payload)
```

Comment:
I was thinking actually turning the input from the web hook into a `BatchReport` object and then just using that -- it's what we use in LLM.